### ISSUE TITLE
Response validation: Automatically coerce date data type to ISO string

### ISSUE THREAD

**Issue by xeoneux** [cdimascio/express-openapi-validator#288]
**Is your feature request related to a problem? Please describe.**
When validating responses, if the data type is a js `Date` object, the validation of type string with format date-time fails.

```js
{
  updatedAt: new Date() // fails
}

{
  updatedAt: "2010-13-30T23:12:35Z" // passes
}
```

**Describe the solution you'd like**
The lib should try to coerce the type to ISO string and then try validation on it.

```js
{
  updatedAt: new Date() // should pass
}
```

**Describe alternatives you've considered**
I have to programmatically search through all the object to find a date data type and call the `toISOString` method on it to pass the output validation.

**Additional context**
None

**Comment by cdimascio** [comment]
@xeoneux is this occurring with response validation?

**Comment by xeoneux** [comment]
This is with response validation, didn't check the request part.

**Comment by cdimascio** [comment]
@xeoneux I did some investigation here. This is a bit tricky to solve. I'm not sure a satisfactory solution exists within Ajv.

I tried to create a custom Ajv keyword. A keyword will enable me to manipulate the `Date` object and convert it to a string e.g. `toISOString()`, The problem is described [here](https://github.com/epoberezkin/ajv/issues/141#issuecomment-270692820) and [here](https://github.com/epoberezkin/ajv/issues/192) note:  
> The only caveat at the moment is that custom keywords will be validated AFTER standard keywords in the same schema object, ...

AFTER is the issue. Ultimately, validation will fail despite modifying the data because an `object` is being passed to a schema that expects a `string` and that validation will fail prior to modifying the data. There is no current way to modify the data prior to validation. 

I certainly agree that using `new Date()` in this context is convenient, but there does not seem to be an obvious solution, other than assigning e.g. `updateAt` to a string e.g. `new Date().toISOString()`

The following is the keyword I was experimenting with
```
    // if this keyword is set, convert a javascript Date to its iso string
    // TODO validate the date, use Date.parse
    ajv.addKeyword('x-eov-date-as-iso-string', {
      modifying: true,
      compile: function (sch, parentSchema) {
        const match = (<any>parentSchema).format === 'date-time' && sch;
        return match
          ? function(value, objectKey, object, key) {
              object[key] = new Date(value).toISOString();
              return true;
            }
          : function() => return true;
      },
    });
```

**Comment by cdimascio** [comment]
keeping this open, perhaps folks have alternative ideas

**Comment by electrotype** [comment]
I have the same problem and found this, which is related: https://github.com/cdimascio/express-openapi-validator/issues/246

Still a workaround, but hey.

**Comment by kapouer** [comment]
You can work around the addKeyword issue with https://github.com/pageboard/server/blob/876236cf0603e4a31b15d793605195008c350648/packages/api/index.js#L70-L100

**Comment by electrotype** [comment]
Do you accept bounties on issues? I could put 50$US for a fix on this one.

This is the only blocking issue for us. We have to `JSON.parse(JSON.stringify(obj))` everything we send, simply for the `date-time` fields to pass the validation.

**Comment by cdimascio** [comment]
@pilerou has provided an option that will help solve this.
please have a look at https://github.com/cdimascio/express-openapi-validator/pull/493 and provide feedback.
thanks!

**Comment by electrotype** [comment]
Feedback given. Thank you all for the effort on this issue.

**Comment by cdimascio** [comment]
this is solved in `v4.10.0`, please give it a try
